src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/PropertyMetadata.cs

```diff
@@ -62,7 +62,10 @@ public PropertyMetadata(IProperty property, Type dbContextType)
 
             IsEnum = property.ClrType.GetTypeInfo().IsEnum;
             IsReadOnly = property.IsReadOnlyBeforeSave;
-            IsAutoGenerated = property.IsStoreGeneratedAlways;
+
+            IsAutoGenerated = ((property.RequiresValueGenerator
+                || property.IsStoreGeneratedAlways)
+                && (property.ValueGenerated != ValueGenerated.Never));
```

Comment:
Thanks @ajcvickers! 

@Andrew-MSFT, @sayedihashimi, @mlorbetske based on the above, looks like we would need to decide on an approach for scaffolding to determine when to not scaffold fields that may be auto-generated. 

In my opinion, having a default behavior to not scaffold fields that follow below pattern should be ok: 
- Require a value generator and 
- Do not have a `[DatabaseGenerated(DatabaseGeneratedOption.None)]`

The users can always go edit the views generated by scaffolding in order to add input to the fields that they want to be set by the application. 

Thoughts?